packages/vhd-lib/src/merge.js

```diff
@@ -21,111 +22,106 @@ export default limitConcurrency(2)(async function merge(
   childPath,
   { onProgress = noop } = {}
 ) {
-  const mergeStatePath = dirname(parentPath) + '/' + '.' + basename(parentPath) + '.merge.json'
-
-  const parentFd = await parentHandler.openFile(parentPath, 'r+')
-  try {
-    const parentVhd = new Vhd(parentHandler, parentFd)
-    const childFd = await childHandler.openFile(childPath, 'r')
-    try {
-      const childVhd = new Vhd(childHandler, childFd)
-
-      let mergeState = await parentHandler.readFile(mergeStatePath).catch(error => {
-        if (error.code !== 'ENOENT') {
-          throw error
-        }
-        // no merge state in case of missing file
-      })
-
-      // Reading footer and header.
-      await Promise.all([
-        parentVhd.readHeaderAndFooter(
-          // dont check VHD is complete if recovering a merge
-          mergeState === undefined
-        ),
-
-        childVhd.readHeaderAndFooter(),
-      ])
-
-      if (mergeState !== undefined) {
-        mergeState = JSON.parse(mergeState)
-
-        // ensure the correct merge will be continued
-        assert.strictEqual(parentVhd.header.checksum, mergeState.parent.header)
-        assert.strictEqual(childVhd.header.checksum, mergeState.child.header)
-      } else {
-        assert.strictEqual(childVhd.header.blockSize, parentVhd.header.blockSize)
-
-        const parentDiskType = parentVhd.footer.diskType
-        assert(parentDiskType === DISK_TYPE_DIFFERENCING || parentDiskType === DISK_TYPE_DYNAMIC)
-        assert.strictEqual(childVhd.footer.diskType, DISK_TYPE_DIFFERENCING)
-      }
+  await Disposable.use(
+    Disposable.all([Vhd.open(parentHandler, parentPath, 'r+'), Vhd.open(childHandler, childPath, 'r')]),
+    async ([parentVhd, childVhd]) => {
```

Comment:
You don't need to use `Disposable.all` for this :slightly_smiling_face: 

```suggestion
    [Vhd.open(parentHandler, parentPath, 'r+'), Vhd.open(childHandler, childPath, 'r')],
    async (parentVhd, childVhd) => {
```